Src/Base/AMReX_Algorithm.H

```diff
@@ -62,6 +63,43 @@ namespace amrex
     {
         return (v < lo) ? lo : (hi < v) ? hi : v;
     }
+
+    template <class T, class F>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    T bisect (T lo, T hi, F f, T tol=1e-12, int max_iter=100)
+    {
+        AMREX_ASSERT_WITH_MESSAGE(hi > lo,
+            "Error - calling bisect but lo and hi don't describe a reasonable interval.");
+
+        T flo = f(lo);
+        T fhi = f(hi);
+
+        if (flo == 0.0_rt) return flo;
+        if (fhi == 0.0_rt) return fhi;
```

Comment:
`T(0)` instead of `0.0_rt`? Should we use enable_if to limit T to floating point?

Reply:
Do you think `std::lower_bound` would work?

Reply:
Never mind. We don't have iterators here.

Reply:
I originally had an enable_if for floating point types here, but it wouldn't compile with Cuda 9.1: https://github.com/AMReX-Codes/amrex/runs/920763725

9.2 or greater was fine, so I suspect it's not our bug, but I couldn't find a work-around other than removing the enable_if.

Reply:
I think instead of `typename std::enable_if<std::is_floating_point<T>{}, int>::type = 0` it should be `typename std::enable_if<std::is_floating_point<T>::value, int>::type = 0`.